src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -1244,20 +1250,33 @@ where
                 }
             });
 
-            remap_input.for_each(|cap, data| {
+            while let Some((_cap, data)) = remap_input.next() {
                 data.swap(&mut remap_trace_buffer);
-                for update in remap_trace_buffer.drain(..) {
-                    timestamper.push_trace_updates(update.into_iter())
-                }
-                cap_set.insert(cap.retain());
-            });
+                remap_updates_stash.append(&mut remap_trace_buffer);
+            }
 
-            let remap_frontier = &frontiers[1];
-            trace!(
-                "reclock({id}) {worker_id}/{worker_count}: remap frontier: {:?}",
-                remap_frontier.frontier()
-            );
-            timestamper.push_upper_update(remap_frontier.frontier().to_owned());
+            // If the remap frontier advanced it's time to carve out a batch that includes all
```

Comment:
ha! when I saw the interface of `ReclockOperator`/`ReclockFollower` and it's `ReclockBack`, I was worried that we would "circumvent" timely's natural batching of messages in the communication layer by sending these larger batches.

This is exactly making sure that we don't do that by unbundling before sending and then re-assembling batches here in the receiver side, nice! 🙌